main.js

```diff
@@ -77,28 +79,28 @@ function deleteIdeaCard(ideaId) {
       var ideaToDelete = ideas[i];
       ideaToDelete.deleteFromStorage();
       ideas.splice(i, 1);
-      displayCards();
     }
   }
 }
 
-function favoriteIdeaCard() {
-
+function favoriteIdeaCard(ideaId) {
+  for (var i = 0; i < ideas.length; i++) {
+    if (ideaId === ideas[i].id) {
+      ideas[i].star = !ideas[i].star;
+      ideas[i].updateIdea();
+    }
+  }
 }
 
 function handleIdeaCardGridClick(event) {
+  var ideaId = Number(event.target.closest('.box-container').id);
   if (event.target.classList.contains('delete-btn')) {
-    var ideaId = Number(event.target.closest('.box-container').id);
     deleteIdeaCard(ideaId);
   }
-}
-
-
-
-
-
-
-
 
+  if (event.target.classList.contains('star-btn')) {
+    favoriteIdeaCard(ideaId);
+  }
 
-//var starredIdeas = [display when user clicks on "Show Starred Ideas"];
+  displayCards();
```

Comment:
Pulled this out because it should happen after either deleting or favoriting cards.